src-tests/suite.lisp

```diff
@@ -78,4 +78,22 @@
          (cloned (rpcq::deserialize (rpcq::serialize original))))
     (is (typep cloned 'rpcq::|RPCRequest|))
     (is (string= (rpcq::|RPCRequest-id| original)     (rpcq::|RPCRequest-id| cloned)))
-    (is (string= (rpcq::|RPCRequest-method| original) (rpcq::|RPCRequest-method| cloned)))))
+    (is (string= (rpcq::|RPCRequest-method| original) (rpcq::|RPCRequest-method| cloned))))
+
+  (let* ((warning (make-instance 'rpcq::|RPCWarning|
+                                 :|body| "The warning string."
+                                 :|kind| "The type of the warning raised."))
+         (original (make-instance 'rpcq::|RPCError|
+                                  :|error| "The error message."
+                                  :|id| "The RPC request id."
+                                  :|warnings| `#(,warning)))
+         (cloned (rpcq::deserialize (rpcq::serialize original))))
+    (is (typep cloned 'rpcq::|RPCError|))
+    (let ((cloned-warnings (rpcq::|RPCError-warnings| cloned)))
+      (is (typep cloned-warnings 'vector))
+      (is (= (array-total-size cloned-warnings) (array-total-size cloned-warnings)))
```

Comment:
Maybe meant to check against `original` warnings size?

Reply:
also use `length`